cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionUtils.java

```diff
@@ -495,23 +495,24 @@ public static AlterPartitionReassignmentsResult maybeStopPartitionReassignment(A
   }
 
   /**
-   * Checks whether the topicPartitions of the execution tasks in the given subset is indeed a subset of the given set.
+   * Find the tasks that need to be re-executed from all candidate inter-broker replica movement tasks.
    *
-   * @param set The original set.
-   * @param subset The subset to validate whether it is indeed a subset of the given set.
-   * @return {@code true} if the topicPartitions of the given subset constitute a subset of the given set, {@code false} otherwise.
+   * @param partitionsInMovement The partitions that are being moved in kafka cluster
+   * @param candidateTasksToReexecute The candidate inter-broker replica movement tasks to re-execute.
+   * @return the set of inter-broker replica movement tasks to re-execute
    */
-  public static boolean isSubset(Set<TopicPartition> set, Collection<ExecutionTask> subset) {
-    boolean isSubset = true;
-    for (ExecutionTask executionTask : subset) {
+  public static List<ExecutionTask> getInterBrokerReplicaTasksToReexecute(Set<TopicPartition> partitionsInMovement,
+                                                                                Collection<ExecutionTask> candidateTasksToReexecute) {
+    List<ExecutionTask> tasksToReexecute = new ArrayList<>();
+
+    for (ExecutionTask executionTask : candidateTasksToReexecute) {
       TopicPartition tp = executionTask.proposal().topicPartition();
-      if (!set.contains(tp)) {
-        isSubset = false;
-        break;
+      if (!partitionsInMovement.contains(tp)) {
+        tasksToReexecute.add(executionTask);
       }
     }
 
-    return isSubset;
+    return tasksToReexecute;
```

Comment:
Maybe we can move logging from `maybeReexecuteInterBrokerReplicaTasks` to here -- i.e. in case `tasksToReexecute` is not empty, we can log both (1) `tasksToReexecute` and (2) `partitionsInMovement` (which is the controller-side movements)?

Reply:
I added the logs in the latest commit. Just not sure if a WARN log makes more sense here, as it seems like not expected to happen if everything works fine?